### PR TITLE
build(deps): Downgrade pydantic again

### PR DESCRIPTION
**Pull request by def-**
Reverts https://github.com/MaterializeInc/materialize/pull/30911

Also upgrades pdoc, but unfortunately newest version of pdoc still fails

Seen failing in https://buildkite.com/materialize/deploy/builds/17206#019414f3-9a6e-4a27-b030-76d435115a0d

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: test/sqllogictest/sqlite
Comment (antiguru) [inline test/sqllogictest/sqlite:1]: Seems unrelated?

**Review by antiguru — APPROVED** [review]
Lgtm, but there seems to be one unrelated change. Disabled auto merge.

File: test/sqllogictest/sqlite
Comment (def-) [reply test/sqllogictest/sqlite:1]: Indeed, I'll have to check why slt changed again, but shouldn't be done in this PR

Comment (def-) [reply test/sqllogictest/sqlite:1]: Actually it was accidentally changed in https://github.com/MaterializeInc/materialize/pull/29159, this change here just undoes it again. I have to say submodules are easy to misuse. So I'd argue while unrelated, this change doesn't hurt.
